src/test/java/org/broadinstitute/hellbender/utils/spark/SparkUtilsShardUnitTest.java

```diff
@@ -0,0 +1,355 @@
+            ImmutableList.of(new SAMSequenceRecord("1", 100), new SAMSequenceRecord("2", 50)));
+
+    @Test
+    public void testSingleContig() throws IOException {
+        JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+
+        //                      1                   2
+        //    1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7
+        //   [-----]
+        //           [-----]
+        //               [-----]
+        // ---------------------------------------------------------
+        //               [-----]
+        //               [-----]
+        //               [-----]
+        // ---------------------------------------------------------
+        //               [-----]
+        //                       [-----]
+        //                         [-----]
+        // ---------------------------------------------------------
+        //                                   [-----]
+        //                                           [-----]
+        //                                                   [-----]
+        // ---------------------------------------------------------
+        //   [-----------------]                                     <- per-partition read extents
+        //               [-----]
+        //               [---------------]
+        //                                   [---------------------]
+        // ---------------------------------------------------------
+        //
+        //     [-----]                                                 <- intervals
+        //                 [---------]
+        //                       [-----------------------]
+        //
+        //                      1                   2
+        //    1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7
+
+        // maxEndPartitionIndexes: [2, 1, 3, 3]
+
+        JavaRDD<TestRead> reads = ctx.parallelize(ImmutableList.of(
+                new TestRead(1, 3), new TestRead(5, 7), new TestRead(7, 9),
+                new TestRead(7, 9), new TestRead(7, 9), new TestRead(7, 9),
+                new TestRead(7, 9), new TestRead(11, 13), new TestRead(12, 14),
+                new TestRead(17, 19), new TestRead(21, 23), new TestRead(25, 27)
+        ), 4);
+
+        List<SimpleInterval> intervals = ImmutableList.of(
+                new SimpleInterval("1", 2, 4),
+                new SimpleInterval("1", 8, 12),
+                new SimpleInterval("1", 11, 22));
+
+        List<ShardBoundary> shardBoundaries = intervals.stream().map(si -> new ShardBoundary(si, si)).collect(Collectors.toList());
+
+        JavaPairRDD<Locatable, Integer> readsPerInterval =
+                SparkUtils.shard(ctx, reads, TestRead.class, sequenceDictionary, shardBoundaries, false)
+                .flatMapToPair(new CountOverlappingReadsFunction());
+        assertEquals(readsPerInterval.collectAsMap(), ImmutableMap.of(intervals.get(0), 1, intervals.get(1), 7, intervals.get(2), 4));
```

Comment:
Declare the Map of expected values separately above, to make it clearer here that you're asserting equality against the same values in both the shuffle and non-shuffle cases. Also applies to the tests below.